### PR TITLE
freac, BoCA, smooth: new port

### PR DESCRIPTION
**Pull request by Janosch**
#### Description
New port for [fre:ac](https://www.freac.org/) audio converter and its dependencies [BoCA](https://github.com/enzo1982/BoCA) and [smooth](https://github.com/enzo1982/smooth).

###### Tested on

macOS 11.1 20C69
Xcode 12.3.0.0.1.1607026830

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on Trac with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by Janosch** [comment]
The azure pipeline fails because it is trying to install the ports in the wrong order. The dependency graph for freac is 

freac -> BoCA -> smooth 

but the script tries to install first BoCA then smooth.

Can this be ignored or do I really have to submit the individual ports one after another?

File: audio/freac/Portfile
Comment (reneeotten) [inline audio/freac/Portfile:12]: can `openmaintainer` be added here?

**Review by reneeotten — CHANGES_REQUESTED** [review]
@Janosch - thanks for the new ports! I just had a cursory look and listed some things that I noticed. I see that there is quite a bit going on in the `post-destroot` phase of these ports... I haven't tried to build these ports and am not sure whether all of that is required and/or cannot be done during the build phase. So a review from someone else is likely needed here before merging.

File: aqua/smooth/Portfile
Comment (Janosch) [reply aqua/smooth/Portfile]: Yes. Done.

File: aqua/smooth/files/patch-fix-libxml2-includes.diff
Comment (Janosch) [reply aqua/smooth/files/patch-fix-libxml2-includes.diff]: Got it. I am using the prefix variable instead, which is already present in the Makefiles.

File: audio/BoCA/Portfile
Comment (Janosch) [reply audio/BoCA/Portfile]: Done.

File: audio/freac/Portfile
Comment (Janosch) [reply audio/freac/Portfile]: Done.

File: aqua/smooth/Portfile
Comment (Janosch) [reply aqua/smooth/Portfile]: Done.

**Comment by Janosch** [comment]
> @Janosch - thanks for the new ports! I just had a cursory look and listed some things that I noticed. I see that there is quite a bit going on in the `post-destroot` phase of these ports... I haven't tried to build these ports and am not sure whether all of that is required and/or cannot be done during the build phase. So a review from someone else is likely needed here before merging.

I fixed the portfiles according to your comments. Changing the references and IDs of the dylibs could also be done by patching the Makefile, but doing it afterwards with install_tool_name is more robust in regard of future changes to the Makefile. 

Furthermore, I removed the git dependency and also removed _--environment-overrides_ from build and destroot args. Both are leftovers which are not needed anymore.

**Review by mf2k — APPROVED** [review]

File: aqua/smooth/Portfile
Comment (mf2k) [inline aqua/smooth/Portfile:31]: Please sort these alphabetically

File: aqua/smooth/Portfile
Comment (mf2k) [inline aqua/smooth/Portfile:30]: We are no longer using jpeg and have switched to libjpeg-turbo. This is how it should be written now:

`path:include/turbojpeg.h:libjpeg-turbo`

File: audio/boca/Portfile
Comment (mf2k) [inline audio/boca/Portfile:26]: Because of this, you need to review and implement [UsingTheRightCompiler](https://trac.macports.org/wiki/UsingTheRightCompiler)

**Review by mf2k — CHANGES_REQUESTED** [review]

File: aqua/smooth/Portfile
Comment (mf2k) [inline aqua/smooth/Portfile:50]: I'm surprised this works. You should use an absolute path to install_name_tool.

File: aqua/smooth/Portfile
Comment (mf2k) [inline aqua/smooth/Portfile:54]: Same here.

**Review by mf2k — CHANGES_REQUESTED** [review]

File: audio/boca/Portfile
Comment (reneeotten) [reply audio/boca/Portfile:26]: that does apply to all three ports here, you might want to use the `makefile` PG as that will likely set some of the things for you correctly. But yes, Frank is right: please make sure these ports all use the correct compiler.

File: aqua/smooth/Portfile
Comment (Janosch) [reply aqua/smooth/Portfile:50]: Hmm, I dont understand. The usage of install_name_tool is 
`install_name_tool [-change old new] input`
I am using absoluth paths to reference the input and to specify the new install name, but for `old` I just have to use the current install name. If I specify something else here install_name_tool will fail because it does not know what to change.

File: aqua/smooth/Portfile
Comment (Janosch) [reply aqua/smooth/Portfile]: Done. I also removed git as it is not a dependency. I added this in the first place because upstream is a github project, but now I realized that I dont use git to fetch the sources.

Comment (Janosch) [reply aqua/smooth/Portfile]: Done. I also removed git because it turned out that it is not a dependency.

File: aqua/smooth/Portfile
Comment (Janosch) [reply aqua/smooth/Portfile]: Done.

File: audio/boca/Portfile
Comment (Janosch) [reply audio/boca/Portfile]: @mf2k Ok, I decided to remove `use_configure no` then. The configure phase does not create any makefiles but if it helps preserving the compiler selection I will keep the configure phase.

@reneeotten I added the makefile PG to all ports. The only feature I use is `makefile.prefix_name` though.

File: aqua/smooth/Portfile
Comment (Janosch) [reply aqua/smooth/Portfile]: Done.

File: aqua/smooth/Portfile
Comment (Janosch) [reply aqua/smooth/Portfile]: Done.

**Review by mf2k — APPROVED** [review]

File: audio/boca/Portfile
Comment (reneeotten) [reply audio/boca/Portfile]: if you use the `makefile` PG then it will automatically set `use_configure no`; you cannot run `./configure` since it doesn't exist for these ports. It's just that the `makefile` PG attempts to set the compilers and flags correctly so that if there is a "standard" Makefile that they can be used.

Comment (Janosch) [reply audio/boca/Portfile]: Ok after reading through makefile-1.0.tcl I think I got it. I am marking this point as resolved as the makefile PG will set the compiler envs the "macports way".

**Comment by Janosch** [comment]
Thanks for the reviews @mf2k and @reneeotten. I resolved the open issues.

**Comment by reneeotten** [comment]
> Thanks for the reviews @mf2k and @reneeotten. I resolved the open issues.

it does build for me locally, but I don't really know how to use any of it so I couldn't test if it actually works... 

Again, I am not sure whether all the stuff in the `post-destroot` is the best/recommended way of accomplishing this or if it could be done somehow in the `build` phase by changing the Makefile. But that's a bit out of my expertise (perhaps @kencu can give offer his insight?).

One comment remaining from me: since these are three new ports, there should be one commit for each port. If you can take care of that I'll merge this tomorrow unless others have comments/suggestions here. If there are things to be fixed I'm sure you'll be happy to take care of that later!

**Comment by Janosch** [comment]
@reneeotten Well for testing you just need to start the freac app, for a more thorough test maybe insert an audio CD and rip it.

I will have a look if I can alter the install name of the dynamic libraries in the Makefile instead of in post-destroot phase, but I dont want to spend too much time on that.

I can make three seperate PRs for this but I'd like to point out that seperating those three ports in seperate PRs makes it really cumbersome for developing and reviewing. I will consider adding smooth and boca as subports of freac instead, as boca and smooth will almost certainly only be used by freac now and in the future.

**Comment by mf2k** [comment]
These portfiles are different enough that leaving them separate seems to make the most sense in this case. sub-ports should be used to remove duplicate code in similar portfiles. Please do make the commits separate for each as @reneeotten requested.

**Comment by Janosch** [comment]
@reneeotten @mf2k I did split up the commit into three commits one for each port. Also, I added patches which fixes the dylib install names for the _smooth_ and _boca_ port. Therefore, calling `install_name_tool` in post-destroot became obsolete for those ports.

I hope this is ready to merge now. If you still see issues I should fix let me know.

**Review by mf2k — APPROVED** [review]

**Comment by Janosch** [comment]
In the latest push I just did some reformatting: giving post-destroot in port freac the correct indent and moving checksums further up (in most ports they are not at the bottom but rather somwhere after the description). Also, I added some line breaks.

**Review by reneeotten — APPROVED** [review]

**Comment by reneeotten** [comment]
thanks @Janosch - in it goes!

**Pull request merged**
